it/src/main/java/org/corfudb/universe/group/Group.java

```diff
@@ -69,11 +69,15 @@
 
     <T extends Node> T getNode(String nodeName);
 
-    interface GroupParams {
+    interface GroupParams<T extends NodeParams> {
         String getName();
 
-        <T extends NodeParams> ImmutableSortedSet<T> getNodesParams();
+        Class<T> getType();
```

Comment:
can we use an enum instead of class?

Reply:
Initially the matching in DockerUniverse was done on CORFU_SERVER, CORFU_CLIENT which is not really a cluster type.

But to answer your question -- no good reason. I will create a separate enum for cluster type.